### PR TITLE
feat: change option of plugins

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

* Change type of `plugins` option from `object` to `array` when using plugin with options : https://github.com/nhn/tui.editor/pull/731/commits/c66e4d8a09524fd78b1ecb80f8e7610150b2d6d6
```js
// Before : Object type
const instance = new Editor({
  plugins: [
    fooPlugin,
    {
      plugin: barPlugin,
      options
    }
  ]
});
```
```js
// After : Array type
const instance = new Editor({
  plugins: [
    fooPlugin,
    [barPlugin, options]
  ]
});
```
* Change usage of `color-syntax` plugin with options : https://github.com/nhn/tui.editor/pull/731/commits/7c037a620cda4f6a5ceaf8c68ed8497ee1950a70
```js
// Before
const instance = new Editor({
  plugins: [colorSyntaxPlugin],
  colorSyntax: options
});
```
```js
// After
const instance = new Editor({
  plugins: [[colorSyntaxPlugin, options]]
});
```
* Update option information of chart plugin : https://github.com/nhn/tui.editor/pull/731/commits/a06ca55c8b7a06bd48a0c31b7582a77f693d79b6
 
---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: plugins/color-syntax/test/plugin.spec.js [hunk 1]
```diff
@@ -2,7 +2,8 @@
  * @fileoverview Test color syntax plugin
  * @author NHN FE Development Lab <dl_javascript@nhn.com>
  */
-import TuiEditor from 'tui-editor/src/js/editor';
+import Editor from 'tui-editor/src/js/editor';
+import colorPicker from 'tui-color-picker';
```
Comment (seonim-ryu) [inline plugins/color-syntax/test/plugin.spec.js]: `tui-color-picker` 모듈 값은 생성자 함수가 아닌 일반 객체이므로 `colorPicker`로 표기함 (의논 필요)

File: plugins/color-syntax/test/plugin.spec.js [hunk 2]
```diff
@@ -22,19 +23,59 @@ describe('colorSyntax', () => {
...
+        usageStatistics: true
+      });
+
+      expect(colorPicker.create).toHaveBeenCalledWith(
+        jasmine.objectContaining({
+          usageStatistics: true
+        })
+      );
+    });
+  });
```
Comment (seonim-ryu) [inline plugins/color-syntax/test/plugin.spec.js]: PR #727 코드 리뷰 반영 내용
> https://github.com/nhn/tui.editor/pull/727#issuecomment-577483599

File: plugins/color-syntax/test/plugin.spec.js [hunk 3]
```diff
@@ -22,19 +23,59 @@ describe('colorSyntax', () => {
...
```
Comment (seonim-ryu) [inline plugins/color-syntax/test/plugin.spec.js]: PR #727 코드 리뷰 반영 내용
> https://github.com/nhn/tui.editor/pull/727#issuecomment-577483599

Comment (seonim-ryu) [inline plugins/color-syntax/test/plugin.spec.js]: PR #727 코드 리뷰 반영 내용
> https://github.com/nhn/tui.editor/pull/727#issuecomment-577483599

File: plugins/color-syntax/test/plugin.spec.js (hunk 1 above)
Comment (seonim-ryu) [inline plugins/color-syntax/test/plugin.spec.js]: `tui-color-picker` 모듈 값은 생성자 함수가 아닌 일반 객체이므로 `colorPicker`로 표기함 (의논 필요)

File: apps/editor/src/js/editor.js [hunk 4]
```diff
@@ -128,7 +128,7 @@ const availableLinkAttributes = ['rel', 'target', 'contenteditable', 'hreflang',
  *     @param {boolean} [options.usageStatistics=true] - send hostname to google analytics
  *     @param {string[]} [options.toolbarItems] - toolbar items.
  *     @param {boolean} [options.hideModeSwitch=false] - hide mode switch tab bar
- *     @param {string[]} [options.plugins] - plugins
+ *     @param {Array.<Function|Array>} plugins - list of plugin function only or plugin function with options
```
Comment (dongwoo-kim) [inline apps/editor/src/js/editor.js]: JSDoc이 튜플을 지원 안하는 것 같은데, 설명이라도 좀 더 명확하게 적어주는 게 좋을 것 같네요. 이대로는 Array에 뭐가 들어갈 지 알기가 어려워서요. 
Array of plugins. A plugin can be either a function or an array in the form of [function, options].

**Pull request merged**